go-zero 是不支持编译 freebsd 下的可执行文件吗

GOOS=freebsd GOARCH=amd64 go build main.go

$ freebsd-version
13.2-RELEASE

$ go run main.go
# github.com/zeromicro/go-zero/core/logx
../../pkg/mod/github.com/zeromicro/go-zero@v1.5.0/core/logx/rotatelogger.go:305:5: undefined: fs.CloseOnExec
../../pkg/mod/github.com/zeromicro/go-zero@v1.5.0/core/logx/rotatelogger.go:368:6: undefined: fs.CloseOnExec

可以确定，go-zero在FreeBSD下是无法编译的。

在Windows平台下交叉编译，也是报同样的错误。

> If go compiles, then go-zero compiles.

--- a/core/fs/files.go
+++ b/core/fs/files.go
@@ -1,4 +1,4 @@
-//go:build linux || darwin
+//go:build linux || darwin || freebsd
 
 package fs
 

--- a/core/proc/goroutines.go
+++ b/core/proc/goroutines.go
@@ -1,4 +1,4 @@
-//go:build linux || darwin
+//go:build linux || darwin || freebsd
 
 package proc
 

--- a/core/proc/goroutines_test.go
+++ b/core/proc/goroutines_test.go
@@ -1,4 +1,4 @@
-//go:build linux || darwin
+//go:build linux || darwin || freebsd
 
 package proc
 

--- a/core/proc/profile.go
+++ b/core/proc/profile.go
@@ -1,4 +1,4 @@
-//go:build linux || darwin
+//go:build linux || darwin || freebsd
 
 package proc
 

--- a/core/proc/shutdown.go
+++ b/core/proc/shutdown.go
@@ -1,4 +1,4 @@
-//go:build linux || darwin
+//go:build linux || darwin || freebsd
 
 package proc
 

--- a/core/proc/shutdown_test.go
+++ b/core/proc/shutdown_test.go
@@ -1,4 +1,4 @@
-//go:build linux || darwin
+//go:build linux || darwin || freebsd
 
 package proc
 

--- a/core/proc/signals.go
+++ b/core/proc/signals.go
@@ -1,4 +1,4 @@
-//go:build linux || darwin
+//go:build linux || darwin || freebsd
 
 package proc
 

--- a/core/proc/signals_test.go
+++ b/core/proc/signals_test.go
@@ -1,4 +1,4 @@
-//go:build linux || darwin
+//go:build linux || darwin || freebsd
 
 package proc
 

--- a/tools/goctl/migrate/cancel.go
+++ b/tools/goctl/migrate/cancel.go
@@ -1,4 +1,4 @@
-//go:build linux || darwin
+//go:build linux || darwin || freebsd
 
 package migrate
 

--- a/tools/goctl/util/pathx/readlink.go
+++ b/tools/goctl/util/pathx/readlink.go
@@ -1,4 +1,4 @@
-//go:build linux || darwin
+//go:build linux || darwin || freebsd
 
 package pathx
 

